docs/api/permissions.rst

```diff
@@ -0,0 +1,331 @@
+        Connection: keep-alive
+        Host: localhost:8888
+        User-Agent: HTTPie/0.9.2
+
+    **Example response**
+
+    .. sourcecode:: http
+
+        HTTP/1.1 200 OK
+        Access-Control-Expose-Headers: Backoff, Retry-After, Alert, Content-Length, Last-Modified, ETag
+        Connection: keep-alive
+        Content-Length: 187
+        Content-Type: application/json; charset=UTF-8
+        Date: Thu, 20 Aug 2015 16:18:48 GMT
+        ETag: "1440087528171"
+        Last-Modified: Thu, 20 Aug 2015 16:18:48 GMT
+        Server: nginx/1.4.6 (Ubuntu)
+
+        {
+            "data": {
+                "id": "fec930f1-4e30-5b1c-2a63-0fafbe508d48",
+                "last_modified": 1440087528171
+            },
+            "permissions": {
+                "write": [
+                    "basicauth:206691a25679e4e1135f16aa77ebcf211c767393c4306cfffe6cc228ac0886b6"
+                ]
+            }
+        }
+
+
+Add a permission
+================
+
+.. http:patch:: /(object url)
+
+    :synopsis: Add principals or permissions to the object.
+
+    **Requires authentication**
+
+    **Example request**
+
+    .. sourcecode:: bash
+
+        $ echo '{"permissions": {"read": ["system.Authenticated"]}}' | \
+          http PATCH https://kinto.dev.mozaws.net/v1/buckets/default/collections/tasks \
+          --auth bob:
+
+    .. sourcecode:: http
+
+        PATCH /v1/buckets/default/collections/tasks HTTP/1.1
+        Accept: application/json
+        Accept-Encoding: gzip, deflate
+        Authorization: Basic Ym9iOg==
+        Connection: keep-alive
+        Content-Length: 52
+        Content-Type: application/json; charset=utf-8
+        Host: kinto.dev.mozaws.net
+        User-Agent: HTTPie/0.8.0
+
+        {
+            "permissions": {
+                "read": [
+                    "system.Authenticated"
+                ]
+            }
+        }
+
+    **Example response**
+
+    .. sourcecode:: http
+
+        HTTP/1.1 200 OK
+        Access-Control-Expose-Headers: Backoff, Retry-After, Alert, Content-Length
+        Connection: keep-alive
+        Content-Length: 188
+        Content-Type: application/json; charset=UTF-8
+        Date: Thu, 20 Aug 2015 16:43:51 GMT
+        ETag: "1440089003843"
+        Last-Modified: Thu, 20 Aug 2015 16:43:23 GMT
+        Server: nginx/1.4.6 (Ubuntu)
+
+        {
+            "data": {
+                "id": "tasks",
+                "last_modified": 1440089003843
+            },
+            "permissions": {
+                "read": [
+                    "system.Authenticated"
+                ],
+                "write": [
+                    "basicauth:206691a25679e4e1135f16aa77ebcf211c767393c4306cfffe6cc228ac0886b6"
+                ]
+            }
+        }
+
+
+Replace or remove permissions
+=============================
+
+.. note::
+
+   The user ID that updates the permissions is always given the ``write``
+   permission, in order to prevent loosing ownership on the object.
```

Comment:
Consider:
> The user ID that *updates* the permissions is always granted the `write` permission. This is in order to prevent accidental loss of ownership on an object.

Out of curiosity, is this a default behaviour that can be changed, or is this a hardcoded behaviour?